packages/react-utilities/src/utils/applyTriggerPropsToChildren.ts

```diff
@@ -11,8 +11,25 @@ export const applyTriggerPropsToChildren = <TTriggerProps>(
   if (typeof children === 'function') {
     return children(triggerProps);
   } else if (children) {
-    return React.cloneElement(onlyChild(children), triggerProps);
+    return cloneTriggerTree(children, triggerProps);
   }
 
   return children;
 };
+
+/**
+ * Clones a React element tree, and applies the given props to the first grandchild that is not
+ * a FluentTriggerComponent or React Fragment (the same element returned by {@link getTriggerChild}).
+ */
+const cloneTriggerTree = <TTriggerProps>(child: React.ReactNode, triggerProps: TTriggerProps): React.ReactElement => {
+  if (!React.isValidElement(child)) {
+    throw new Error(`Component's child must be a single element`);
```

Comment:
with this error in mind, the fragment support seems unnecessary... do we want to enable users to do this ?

```
<>
  <div> This is a useless fragment </div>
</> 
```

Reply:
Updated it to throw for fragments.